### PR TITLE
Support dataTable.header.extend in theme and expose prop

### PR DESCRIPTION
**Pull request by halocline**
Signed-off-by: Matt Glissmann <mdglissmann@gmail.com>

#### What does this PR do?

Enables a caller to specify styling via `dataTable.header.extend` and passes props to a header cell's button.

#### Where should the reviewer start?

StyledDataTable.js, base.js, and /DataTable/Header.js

#### What testing has been done on this PR?

Jest + modified Storybook story DataTable/custom

#### How should this be manually tested?

Storybook story DataTable/custom

#### Any background context you want to provide?

[Design File](https://www.figma.com/file/XEoIWIlLfcARRIFhkpFCch/HPE-Table?node-id=2390%3A13951)

Addressing a DataTable design which calls for a Header Cell's `icons.sortable` to only be shown upon hover, while an icons indicated a sort has been applied to persist. This behavior seemed opinionated and particular to this specific design, so this PR opts for a more generalized solution allowing for that customization to be done via `extend`.

#### What are the relevant issues?

https://github.com/grommet/hpe-design-system/issues/1180

#### Do the grommet docs need to be updated?

Yes. Included.

#### Should this PR be mentioned in the release notes?

Pretty minor.

#### Is this change backwards compatible or is it a breaking change?

Backwards compatible.

### CODE REVIEW

File: src/js/components/DataTable/StyledDataTable.js [hunk 1]
```diff
@@ -105,6 +105,10 @@ const StyledDataTableHeader = styled(TableHeader)`
     width: 100%;
     table-layout: fixed;
   `}
+  ${props =>
+    props.theme.dataTable &&
+    props.theme.dataTable.header &&
+    props.theme.dataTable.header.extend};
```
Comment (ericsoderberghp) [inline src/js/components/DataTable/StyledDataTable.js:111]: Shouldn't this be applied to the `<th />` cells instead of the `<thead />` header element?

**Review by taysea — COMMENTED** [review]
Aside from Eric's comment, I think the rest looks appropriate

File: src/js/components/DataTable/StyledDataTable.js (hunk 1 above)
Comment (halocline) [reply src/js/components/DataTable/StyledDataTable.js]: Done.

**Review by ericsoderberghp — APPROVED** [review]

**Comment by ericsoderberghp** [comment]
Can you look into why the header cells are being styled differently in the DataTable Custom story? Chromatic reported this and it seems like it might be a breaking change.

**Comment by halocline** [comment]
> Can you look into why the header cells are being styled differently in the DataTable Custom story? Chromatic reported this and it seems like it might be a breaking change.

I believe it is because I modified the table header cell text to be bold, which caused the overall table width to grow, slightly displacing the rest of the table content. I will remove that styling change, and re-run Chromatic to confirm.

**Comment by halocline** [comment]
> > Can you look into why the header cells are being styled differently in the DataTable Custom story? Chromatic reported this and it seems like it might be a breaking change.
> 
> I believe it is because I modified the table header cell text to be bold, which caused the overall table width to grow, slightly displacing the rest of the table content. I will remove that styling change, and re-run Chromatic to confirm.

@ericsoderberghp I removed some of the additional styling I had added to the Custom story (just left in the hover styling) and the Chromatic snapshots no longer show a diff.

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**